server/apiserver/argoserver.go

```diff
@@ -156,11 +160,31 @@ func (as *argoServer) Run(ctx context.Context, port int) {
 	go func() { as.checkServeErr("grpcServer", grpcServer.Serve(grpcL)) }()
 	go func() { as.checkServeErr("httpServer", httpServer.Serve(httpL)) }()
 	go func() { as.checkServeErr("tcpm", tcpm.Serve()) }()
-	log.Infof("Argo Server started successfully on address %s", address)
+	log.Infof("Argo Server started successfully on address http://locahost%s", address)
+	openbrowser("http://localhost" + address)
```

Comment:
this won't work in hosted mode, I think you should have to do

`argo server --browser`

I'd move this all the `cmd/argo/server.go`